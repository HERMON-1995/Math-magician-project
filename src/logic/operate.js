import Big from 'big.js';

class OperationError extends Error {
  constructor(message) {
    super(message);
    this.name = 'OperationError';
  }
}

export default function operate(numberOne, numberTwo, operation) {
  const one = Big(numberOne);
  const two = Big(numberTwo);
  let result;

  switch (operation) {
    case '+':
      result = one.plus(two);
      break;
    case '-':
      result = one.minus(two);
      break;
    case 'x':
      result = one.times(two);
      break;
    case 'รท':
      if (two.eq(0)) {
        throw new OperationError("Can't divide by 0.");
      }
      result = one.div(two);
      break;
    case '%':
      if (two.eq(0)) {
        throw new OperationError("Can't find modulo as can't divide by 0.");
      }
      result = one.mod(two);
      break;
    default:
      throw new OperationError(`Unknown operation '${operation}'`);
  }

  return result.toString();
}
