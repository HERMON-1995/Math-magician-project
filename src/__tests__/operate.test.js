import operate from '../logic/operate';

describe('operations in calculator', () => {
  test('multiply 3*3 to equal 9', () => {
    const result = operate(3, 3, 'x');
    expect(result).toBe('9');
  });
  test('division of 10/2', () => {
    const divide = operate(10, 2, 'รท');
    expect(divide).toBe('5');
  });
  test('substraction to substract 10-2', () => {
    const substract = operate(2, 3, '-');
    expect(substract).toBe('-1');
  });
  test('addition of 10+2', () => {
    const addition = operate(10, 2, '+');
    expect(addition).toBe('12');
  });
  test('modulos', () => {
    const modulos = operate(5, 2, '%');
    expect(modulos).toBe('1');
  });
  test('multiply 10*2', () => {
    const result = operate(10, 2, 'x');
    expect(result).toBe('20');
  });
});
