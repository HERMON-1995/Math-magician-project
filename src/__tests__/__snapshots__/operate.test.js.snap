// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`operate function should return correct result 1`] = `"5"`;
